### ISSUE TITLE
ListGroupItem: href to internal routes not linking properly

### ISSUE THREAD

**Issue by thomashusken** [facultyai/dash-bootstrap-components#877]
Before opening:

- [Search for duplicate or closed issues](https://github.com/facultyai/dash-bootstrap-components/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Read the [contributing guidelines](https://github.com/facultyai/dash-bootstrap-components/blob/main/.github/CONTRIBUTING.md)

Please fill out the below information as much as possible.

- dash version: `#2.5.1`
- dash-bootstrap-components version: `#1.2.0`
- components affected by bug: `dbc.ListGroupItem`, possibly other components as well

### What is happening?

When passing an internal route, such as `/foo`, as href in a ListGroupItem, the resulting item doesn't actually redirect to that route. 

### What should be happening?

Clicking on the ListGroupItem should redirect to the `/foo` path. The issue can be circumvented by wrapping a `html.A()` link around the ListGroupItem and providing the href there. This is still a workaround, however. 

### Code

```python
from dash import Dash, html
from flask import Flask

import dash_bootstrap_components as dbc

flask_app = Flask(__name__)

layout = html.Div([
    dbc.ListGroup([
        dbc.ListGroupItem("Link to foo", href="/foo")
    ])
])

dash = Dash(
    __name__,
    server=flask_app,
    url_base_pathname='/',
    suppress_callback_exceptions=True,
    compress=False,
    external_stylesheets=[dbc.themes.BOOTSTRAP],
    serve_locally=True,
)

dash.layout = layout


@dash.server.route("/foo")
def foo():
    return "foo"


if __name__ == "__main__":
    dash.run(debug=True, port=8000)
```

**Comment by tcbegley** [comment]
Hi @thomashusken, thanks for opening this and sorry it took me a little while to get around to responding.

TL;DR: try setting `external_link=True` on the `ListGroupItem`.

The behaviour you're observing is to allow `ListGroupItem` to be used with `dcc.Location` for internal navigation, where clicking the link prevents the browser from actually following the link, and instead updates the address bar and triggers any callbacks that are using `dcc.Location.pathname` or similar as an input. This allows you to navigate between "pages" of the Dash app while maintaining state and not reloading javascript bundles. In effect the Dash app works like a [Single-Page Application (SPA)](https://en.wikipedia.org/wiki/Single-page_application).

Of course, as in your case, you may actually want to access a route on your flask server by having the browser actually load the page. In such cases, you should set `external_link=True` on the `ListGroupItem`. The component does actually try to set this intelligently, e.g. if you give it a path like `/internal` it will create a component that behaves like `dcc.Link`, on the other hand if you give it `https://google.com` it will give you a component that behaves like `html.A`.

I think to help prevent other people getting confused by this in the future, I will take a pass at updating / clarifying the docstrings for `external_link`.

**Comment by thomashusken** [comment]
Hi @tcbegley!

Thanks for the clear answer! I guess I misunderstood what is considered an external link in this case, and can confirm that setting `external_link = True` solves the problem for me. Closing this issue accordingly :)
